src/ui/interface.js

```diff
@@ -72,30 +74,103 @@ module.exports = function Interface(options) {
       $('.choose-prompt').hide();
       $("#save-modal-btn").show();
       $("#save-zone").show();
-      let reader = new FileReader();
-      reader.onload = function onReaderLoad(event) {
-        var img;
-        img = new Image();
-        img.onload = function onImageLoad() {
-          options.processor.updateImage(this);
+      if ((this.files[0].type == 'image/jpeg')||(this.files[0].type == 'image/png')) {
+        let reader = new FileReader();
+        reader.onload = function onReaderLoad(event) {
+          var img;
+          img = new Image();
+          img.onload = function onImageLoad() {
+            options.processor.updateImage(this);
+          };
+          return img.src = event.target.result;
         };
-        return img.src = event.target.result;
-      };
-      reader.readAsDataURL(this.files[0]);
-      $('#preset-modal').modal('show');
+        reader.readAsDataURL(this.files[0]);        
+      }else{
```

Comment:
There's a little bit of formatting cleanup we could do here maybe?

Reply:
Yes alot has to be done here. Thank you,  I keep on forgetting
1. Show image after Webrtc is closed.
2. Hide Video controls when image is selected.

Reply:
**Image files are unable to load after camera stream.**
  
@jywarren , please i can't seem to diagnose the source of this unexpected behavior. Could you help me please?

https://user-images.githubusercontent.com/24577149/181921156-3801abe2-c445-4903-855a-f1611257ceef.mp4

Reply:
OK, so i think when we start video mode, we begin a loop to fetch images from the video element. When we try to go back to upload mode, we need to find and halt that loop. Let me look...

Reply:
Yes, we do it in this function:

https://github.com/publiclab/infragram/blob/f15add6ae6057bd99b4b1b930449f75ca974270e/src/Infragram.js#L34-L45

And that function is called from here:

https://github.com/publiclab/infragram/blob/577ff122edd6918891a84ea9d9df221585fdc901/src/ui/interface.js#L89-L97

So we have to save the `setInterval` object and stop it with `clearInterval()`. Intervals are a little weird but they are just repeated timers. You can read how they work here! https://developer.mozilla.org/en-US/docs/Web/API/setInterval#return_value There are good examples lower on that page that you can model to use here. But, I think we could also solve this in a separate PR - i don't think it's your code that's causing it, it's already the case in the main branch!

And, solving it may be a little more complex, we have to think carefully about where we store the interval return value so that we can make a `stopVideo()` function, probably? I can help you figure that out too but we might as well move this whole comment thread to its own issue, what do you think?

Reply:
Moved this thread to https://github.com/publiclab/infragram/issues/441!